Pod/Classes/JNKeychain.m

```diff
@@ -22,14 +24,31 @@ @implementation JNKeychain
 + (NSMutableDictionary *)getKeychainQuery:(NSString *)key
 {
     // see http://developer.apple.com/library/ios/#DOCUMENTATION/Security/Reference/keychainservices/Reference/reference.html
-    return [@{(__bridge id)kSecClass            : (__bridge id)kSecClassGenericPassword,
+    NSMutableDictionary *query = [@{(__bridge id)kSecClass            : (__bridge id)kSecClassGenericPassword,
               (__bridge id)kSecAttrService      : key,
               (__bridge id)kSecAttrAccount      : key,
               (__bridge id)kSecAttrAccessible   : (__bridge id)kSecAttrAccessibleAfterFirstUnlock
               } mutableCopy];
+    if ([__accessGroup length] > 0) {
+        query[(__bridge id)(kSecAttrAccessGroup)] = __accessGroup;
+    }
+    return query;
 }
 
 #pragma mark - Public
+
++ (void)setAccessGroup:(NSString *)accessGroup {
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        __accessGroup = [[NSMutableString alloc] init];
+    });
+    if (accessGroup == nil) {
```

Comment:
If __accessGroup were an NSString we could remove the `dispatch_once` and just do:

    __accessGroup = accessGroup ?: @"";

What do you think?